VirtualFinland.UserAPI/src/VirtualFinland.UsersAPI/Data/Repositories/LanguageRepository.cs

```diff
@@ -1,21 +1,42 @@
-using System.Globalization;
+using System.Text.Json;
 using VirtualFinland.UserAPI.Models.Repositories;
 
 namespace VirtualFinland.UserAPI.Data.Repositories;
 
 public class LanguageRepository : ILanguageRepository
 {
+    private readonly IHttpClientFactory _httpClientFactory;
+    private readonly string _languagesUrl;
+    private List<Language>? _languages;
 
-    public Task<List<Language>> GetAllLanguages()
+
+    public LanguageRepository(IConfiguration configuration, IHttpClientFactory httpClientFactory)
+    {
+        _httpClientFactory = httpClientFactory;
+        _languagesUrl = Environment.GetEnvironmentVariable("CODE_SET_LANGUAGES") ?? configuration["ExternalSources:ISO639Languages"]; ;
+    }
+
+    public async Task<List<Language>> GetAllLanguages()
     {
-        return Task.FromResult(CultureInfo.GetCultures(CultureTypes.AllCultures)
-            .Where(o => !string.IsNullOrEmpty(o.Name) && o.Name == o.TwoLetterISOLanguageName)
-            .Select(o => new Language(o.ThreeLetterISOLanguageName,
-                o.DisplayName,
-                o.EnglishName,
-                o.NativeName,
-                o.TwoLetterISOLanguageName,
-                o.ThreeLetterISOLanguageName))
-            .ToList());
```

Comment:
Olikohan tässä jokin erityinen syy, että käytettiin (tai käytin, olikohan minun tekosia 😅 ) tuota CutureInfo apia 🤔, Noh jos tulee tarvittava dada edelleen uudesta listasta ja homma pelittää niin muy bueno.

Reply:
Dotnetin toi cultureinfo palauttaa hieman rikkaampia käännöksiä kuin nyt codesetsiin tehty vastine, mutta suorittavalta osalta data mätsää eli toiminnallisuus säilyy ennallaan. Tämän olisi toki voinut tehdä niinkin päin että codesets lukee tiedon users-api:lta jolloin olisi säilyny vanha kirjastototeutus ja silti datan eheys/integrity säilynyt, vaan ehkä loogisempi totetuttaa siellä codesetsien päässä.